projects/canopy/src/lib/side-nav/side-nav-bar/side-nav-bar.component.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  AfterContentChecked,
+  ChangeDetectorRef,
+  Component,
+  ContentChildren,
+  forwardRef,
+  HostBinding,
+  HostListener,
+  Input,
+  OnDestroy,
+  QueryList,
+  ViewEncapsulation,
+} from '@angular/core';
+
+import { merge, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { LgSideNavBarLinkDirective } from '../side-nav-bar-item/side-nav-bar-link.directive';
+import { isKeyDown, isKeyLeft, isKeyRight, isKeyUp } from '../../utils/keyboard-keys';
+
+@Component({
+  selector: 'lg-side-nav-bar',
+  templateUrl: './side-nav-bar.component.html',
+  styleUrls: ['./side-nav-bar.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LgSideNavBarComponent implements AfterContentChecked, OnDestroy {
```

Comment:
There is obviously quite a lot of repetition here between this and the [tab-nav-bar](https://github.com/Legal-and-General/canopy/blob/master/projects/canopy/src/lib/tabs/tab-nav-bar/tab-nav-bar.component.ts) component. I wonder if there is a way of creating a more generic approach that works for both, `side-nav` and `tab-nav` are almost variants of the same thing.

```ts
selector: 'lg-side-nav-bar, 'lg-side-nav-bar'',
```

We did set a precedent on the [radio component](https://github.com/Legal-and-General/canopy/blob/master/projects/canopy/src/lib/forms/radio/radio-button.component.ts) where it has the ability to to behave differently, I wonder if this is similar.

Reply:
I think the main thing that sets them apart is the fact that the sidebar content + nav are oriented differently based on screen width (horizontal on desktop, vertical on mobile). Not sure if this is easily achievable with the current tab implementation.

Reply:
I wondered if there is an opportunity to make the component code common, but have separate css files for the side / tab navigation approaches? I think if there was to be lots of code branching in the component this might get ugly, but if most of it is presentational then it could make it more DRY. Happy to offer a bit of support to help on it if needed?

Reply:
Sure, I think it's best we have a three amigos first with you, me, and Owen so we can outline the plan for this.

Reply:
Sounds good, are you ok to schedule something in? Shout if you can't find a time and I can re-arrange things. @owensgit is this all ok from your side?